library/src/math/cancel.rs

```diff
@@ -0,0 +1,171 @@
+///
+/// Or draw two lines (a cross): $ x + cancel(a + b + c, cross: #true) $
+///
+/// Or set its length, relative to the diagonal: $cancel(x, length: #200%)$
+///
+/// Rotating and even making it red and thicker is possible: $ cancel(x, rotate: #30deg, stroke: #{red + 1.5pt}) $
+/// ```
+///
+/// Display: Cancel
+/// Category: math
+#[element(LayoutMath)]
+pub struct CancelElem {
+    /// The content over which the line should be placed.
+    #[required]
+    pub body: Content,
+
+    /// The length of the line, relative to the length of the main diagonal spanning the whole
+    /// element being "cancelled".
+    ///
+    /// Defaults to `{100% + 3pt}`.
+    pub length: Smart<Rel<Length>>,
+
+    /// If the cancel line should be inverted (heading northwest instead of northeast).
+    ///
+    /// Defaults to `{false}`.
+    #[default(false)]
+    pub invert: bool,
+
+    /// If two opposing cancel lines should be drawn, forming a cross over the element.
+    /// Overrides `invert`.
+    ///
+    /// Defaults to `{false}`.
+    #[default(false)]
+    pub cross: bool,
+
+    /// Rotate the cancel line by a certain angle. See the
+    /// [line's documentation]($func/line.angle) for more details.
+    #[default(Angle::zero())]
+    pub rotate: Angle,
+
+    /// How to stroke the cancel line. See the
+    /// [line's documentation]($func/line.stroke) for more details.
+    #[resolve]
+    #[fold]
+    pub stroke: PartialStroke,
+}
+
+impl LayoutMath for CancelElem {
+    fn layout_math(&self, ctx: &mut MathContext) -> SourceResult<()> {
+        let mut body = ctx.layout_frame(&self.body())?;
+
+        let size = body.size();
+        let (width, height) = (size.x, size.y);
+
+        let length = self
+            .length(ctx.styles()) // empirically pleasant default
+            .unwrap_or(Rel::new(Ratio::one(), Abs::pt(3.0).into()))
+            .resolve(ctx.styles());
+
+        // default stroke has 0.5pt for better visuals
+        let stroke = self.stroke(ctx.styles()).unwrap_or(Stroke {
+            paint: TextElem::fill_in(ctx.styles()),
+            thickness: Abs::pt(0.5),
+            ..Default::default()
+        });
+
+        let invert = self.invert(ctx.styles());
+        let cross = self.cross(ctx.styles());
+        let angle = self.rotate(ctx.styles());
+
+        let invert_first_line = !cross && invert;
+
+        let first_line = draw_cancel_line(
+            length,
+            stroke.clone(),
+            invert_first_line,
+            angle,
+            size,
+            self.span(),
+        );
+
+        // the origin of our line is the very middle of the element
+        body.push_frame(Point::new(width / 2.0, height / 2.0), first_line);
+
+        if cross {
+            // draw the second line
+            let second_line = draw_cancel_line(
+                length,
+                stroke,
+                /*invert:*/ true,
```

Comment:
```suggestion
                true,
```
it may slightly improve clarity, but I don't think its worth it to introduce style inconsistencies like this